service/src/main/java/bio/terra/profile/service/profile/flight/create/LinkBillingProfileIdToMrgStep.java

```diff
@@ -0,0 +1,51 @@
+package bio.terra.profile.service.profile.flight.create;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.profile.service.azure.ApplicationService;
+import bio.terra.profile.service.profile.exception.MissingRequiredFieldsException;
+import bio.terra.profile.service.profile.model.BillingProfile;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+
+/** Add a tag that associates a billing profile ID with an azure managed resource group */
+public record LinkBillingProfileIdToMrgStep(
```

Comment:
This is the new step, adding the billing profile ID as a tag to the MRG via a call to new code in ApplicationService.